sdk/java/dagger-java-annotation-processor/src/main/java/io/dagger/annotation/processor/DaggerModuleAnnotationProcessor.java

```diff
@@ -0,0 +1,324 @@
+                ", new $T.WithObjectArguments().withDescription($S)",
+                TypeDef.class,
+                objectInfo.description());
+          }
+          rm.addCode(")"); // end of dag.TypeDef().withObject(
+          for (var fnInfo : objectInfo.functions()) {
+            rm.addCode("\n            .withFunction(")
+                .addCode("\n                dag.function($S,", fnInfo.name())
+                .addCode("\n                    ")
+                .addCode(typeDef(fnInfo.returnType()))
+                .addCode(")");
+            if (isNotBlank(fnInfo.description())) {
+              rm.addCode("\n                    .withDescription($S)", fnInfo.description());
+            }
+            for (var parameterInfo : fnInfo.parameters()) {
+              rm.addCode("\n                    .withArg($S,", parameterInfo.name())
+                  .addCode(typeDef(parameterInfo.type()))
+                  .addCode(")");
+            }
+            rm.addCode(")"); // end of .withFunction(
+          }
+          rm.addCode(")"); // end of .withObject(
+        }
+        rm.addCode(";\n") // end of module instantiation
+            .addStatement("return module.id()");
+
+        var im =
+            MethodSpec.methodBuilder("invoke")
+                .addModifiers(Modifier.PRIVATE)
+                .returns(JSON.class)
+                .addException(Exception.class)
+                .addParameter(JSON.class, "parentJson")
+                .addParameter(String.class, "parentName")
+                .addParameter(String.class, "fnName")
+                .addParameter(
+                    ParameterizedTypeName.get(Map.class, String.class, JSON.class), "inputArgs")
+                .beginControlFlow("try (var jsonb = $T.create())", JsonbBuilder.class);
+        var firstObj = true;
+        for (var objectInfo : annotatedObjects) {
+          if (firstObj) {
+            firstObj = false;
+            im.beginControlFlow("if (parentName.equals($S))", objectInfo.name());
+          } else {
+            im.nextControlFlow("else if (parentName.equals($S))", objectInfo.name());
+          }
+          im.addStatement("$T clazz = Class.forName($S)", Class.class, objectInfo.qualifiedName())
+              .addStatement("var obj = $T.fromJSON(dag, parentJson, clazz)", Convert.class)
+              .addStatement(
+                  "clazz.getMethod(\"setClient\", $T.class).invoke(obj, dag)", Client.class);
+          var firstFn = true;
+          for (var fnInfo : objectInfo.functions()) {
+            var fnReturnClass = Class.forName(fnInfo.returnType());
+            if (firstFn) {
+              firstFn = false;
+              im.beginControlFlow("if (fnName.equals($S))", fnInfo.name());
+            } else {
+              im.nextControlFlow("else if (fnName.equals($S))", fnInfo.name());
+            }
+            var fnBlock =
+                CodeBlock.builder().add("$T fn = clazz.getMethod($S", Method.class, fnInfo.name());
```

Comment:
it should be `fnInfo.qName()` otherwise it should throw a `NoSuchMethodException` when an alias is defined